fcitx-contrib/fcitx5-macos
# restart process on plugin upgrade

Test:
* `tail -f /tmp/Fcitx5.log` and later you should see cleanup log when Fcitx5 quits, like
```sh
Dinstance.cpp:1408] Running save...
Iaddonmanager.cpp:296] Unloading addon spell
Iaddonmanager.cpp:296] Unloading addon chttrans
Iaddonmanager.cpp:296] Unloading addon fullwidth
Iaddonmanager.cpp:296] Unloading addon cloudpinyin
Devent_libuv.cpp:116] UVLoop close: 0
Iaddonmanager.cpp:296] Unloading addon pinyin
Iaddonmanager.cpp:296] Unloading addon punctuation
Iaddonmanager.cpp:296] Unloading addon keyboard
Iaddonmanager.cpp:296] Unloading addon unicode
Iaddonmanager.cpp:296] Unloading addon imeapi
Iaddonmanager.cpp:296] Unloading addon luaaddonloader
Iaddonmanager.cpp:296] Unloading addon pinyinhelper
Iaddonmanager.cpp:296] Unloading addon notifications
Iaddonmanager.cpp:296] Unloading addon clipboard
Iaddonmanager.cpp:296] Unloading addon quickphrase
Iaddonmanager.cpp:296] Unloading addon imselector
Iaddonmanager.cpp:296] Unloading addon macosfrontend
Iaddonmanager.cpp:296] Unloading addon webpanel
Devent_libuv.cpp:116] UVLoop close: -16
Devent_libuv.cpp:125] UVLoop close r2: 0
```
* Clean `~/Library/fcitx5/cache`. Plugins won't be cached anymore, but developers are still able to copy into it manually for slow network.
* Assume you don't have latest chinese-addons installed, then input s in Pinyin you will see 7th candidate 丨, which has comment with the same style.
* In plugin manager, uninstall chinese-addons then Install silently, should auto restart, then clicking penguin icon twice or clicking a client program, you can see Fcitx5 revived, and 丨's comment should have different style.
* Try normal install, and you will be prompted to restart.
* Click restart in IM menu many times to make sure Fcitx5 can revive.

I know upgrade in about window could also make use of this, but would rather not optimize right now. It could be replaced later.
